PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -1,14 +1,468 @@
+                return;
+            }
+
+            if (browserSwitchNotPossible(activity)) {
+                braintreeClient.sendAnalyticsEvent("paypal.invalid-manifest");
+                Exception manifestInvalidError = createBrowserSwitchError();
+                callback.onResult(manifestInvalidError);
+                return;
+            }
+
+            sendPayPalRequest(
+                activity,
+                payPalVaultRequest,
+                configuration,
+                callback
+            );
+        });
+    }
+
+    private void sendPayPalRequest(
+        final FragmentActivity activity,
+        final PayPalRequest payPalRequest,
+        final Configuration configuration,
+        final PayPalFlowStartedCallback callback
+    ) {
+        internalPayPalClient.sendRequest(activity, payPalRequest, (payPalResponse, error) -> {
+            if (payPalResponse != null) {
+                String analyticsPrefix = getAnalyticsEventPrefix(payPalRequest);
+                braintreeClient.sendAnalyticsEvent(String.format("%s.app-switch.started", analyticsPrefix));
+
+                Environment environment;
+                if ("sandbox".equals(configuration.getEnvironment())) {
+                    environment = Environment.SANDBOX;
+                } else if ("production".equals(configuration.getEnvironment())) {
+                    environment = Environment.LIVE;
+                } else {
+                    callback.onResult(new IllegalArgumentException("Invalid PayPal Environment"));
+                    return;
+                }
+
+                // Start PayPalCheckout flow
+                PayPalCheckout.setConfig(
+                    new CheckoutConfig(
+                        activity.getApplication(),
+                        configuration.getPayPalClientId(),
+                        environment,
+                        payPalRequest.getNativeConfig().getReturnUrl()
+                    )
+                );
+
+                registerCallbacks(configuration, payPalRequest, payPalResponse);
+
+                PayPalCheckout.startCheckout(createOrderActions -> {
+                    if (payPalRequest instanceof PayPalCheckoutRequest) {
+                        createOrderActions.set(payPalResponse.getPairingId());
+                    } else if (payPalRequest instanceof PayPalVaultRequest) {
+                        createOrderActions.setBillingAgreementId(payPalResponse.getPairingId());
+                    } else {
+                        callback.onResult(new Exception("Unsupported action type"));
+                    }
+                });
+                callback.onResult(null);
+            } else {
+                callback.onResult(error);
+            }
+        });
+    }
+
+    private void registerCallbacks(
+        final Configuration configuration,
+        final PayPalRequest payPalRequest,
+        final PayPalResponse payPalResponse
+    ) {
+        PayPalCheckout.registerCallbacks(
```

Comment:
@tdchow do you think we should allow the user to pass their own callbacks here? Originally I was thinking we should but we need to control the `internalPayPalClient.tokenize` call here

Reply:
I believe this would be up to the BT team on what their external interface would look like. Would we want to keep this new native PayPal interface matching the others?

Reply:
Ok, we can defer adding the callbacks for now then